### PR TITLE
Switch XLA to new operator registration API.

### PR DESCRIPTION
**Pull request by ezyang**
Companion patch for https://github.com/pytorch/pytorch/pull/35061 . I don't know if it works or not. Don't land before that PR lands.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Comment by ailzhang** [comment]
@dlibenzi FYI this will be switching us to use the new op registration API.

File: scripts/gen.py [hunk 1]
```diff
@@ -961,12 +961,12 @@ def generate_registrations(fgens, overrides):
...
+      overload = fgen.funsig[:pos] + ' (*)' + fgen.funsig[pos:]
       code += (
-          '  .op(torch::RegisterOperators::options().schema("{}")\n      '
-          '.impl_unboxedOnlyKernel<{}, &{}>(at::DispatchKey::XLATensorId)\n'
-          '      .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA))\n'.format(
-              fgen.aten_sig, fgen.funsig, override_fn, override_fn,
-              fgen.aten_sig))
+          '  .impl("{}", torch::dispatch(at::DispatchKey::XLATensorId, '
+          'at::CppFunction::makeUnboxedOnly(static_cast<{}>(&{}))))\n'.format(
+              fgen.aten_sig.split("(")[0], overload, override_fn))
```
Comment (dlibenzi) [inline scripts/gen.py]: We use '' as string constant delimiters.

**Comment by dlibenzi** [comment]
@ailzhang Be sure to clean this up ... there seem to be master's commits in the log 😉

**Comment by ailzhang** [comment]
(sorry I messed up the history here due to a merge conflict with master :(

**Pull request merged**